server/app/routers.py

```diff
@@ -59,17 +68,28 @@ async def update_graph(
 async def search_graph(
     query: SPARQLQuery,
 ) -> SearchResponse:
-    """Execute SPARQL search query and return a response in JSON-LD format."""
-    # result = fuseki.read_query(query)
-    # if type(result) is Bindings:
-    #     bindings = []
-    #     for item in result.bindings:
-    #         new_item = {}
-    #         for key in item:
-    #             if hasattr(item[key], "value") and hasattr(item[key], "type"):
-    #                 new_item[key] = {"value": item[key].value, "type": item[key].type}
-    #         bindings.append(new_item)
-    #     return bindings
-    # else:
-    #     return str(result)
-    return EMPTY_SEARCH_RESPONSE  # Fix code above, then delete this line
+    """Execute SPARQL search query and return a response in JSON format."""
+    result = fuseki.read_query(query)
+
+    if type(result) is Bindings:
+        bindings = []
+        for item in result.bindings:
+            new_item: Dict[str, Any] = {}
+            for key in item:
+                new_item[key] = {}
+                for property in item[key].__dict__:
+                    if (
+                        property != "variable"
+                        and item[key].__dict__[property] is not None
+                    ):
+                        new_item[key][property] = item[key].__dict__[property]
+                        if property == "lang":
+                            new_item[key]["xml:lang"] = new_item[key].pop("lang")
+            bindings.append(new_item)
+
+        return SearchResponse(
+            head=ResponseHead(vars=result.head["vars"]),
+            results=ResponseResults(bindings=bindings),
+        )
+
+    return EMPTY_SEARCH_RESPONSE
```

Comment:
If we come to this line, is it a case of some error? If yes, it's much better to set a proper HTTP response code and return provide error information.

Reply:
I will update it with proper HTTP responses. But first I'll make sure that this version works on the cluster as well.